File: 11INFORMATIKA/DATASCIENCE/11KOM1/meet18/car.cpp
#include <iostream>
#include <string>
using namespace std;

struct Vehicle {
    string type;

    Vehicle(string t) : type(t) {}
};

struct Car : Vehicle {
    string make;
    string model;
    int year;
    int odometer;

    Car(string ma, string mo, int y) : Vehicle("car"), make(ma), model(mo), year(y), odometer(0) {}

    void increment_odometer(int n = 1){
        odometer += n;
    }

    void describe(){
        cout << "this is a " << type << ", made by " << make << endl;
    }

};

int main(){
    Car my_tesla("Tesla", "S Series", 2023);
    my_tesla.describe();
    return 0;
}

File: 11INFORMATIKA/DATASCIENCE/11KOM1/meet02/meet02i/car.py
class Vehicle:

    def __init__(self, type):
        self.type = type
    
    def move(self):
        print(f"The {self.type} is moving ...")

class Boat(Vehicle):

    def __init__(self):
        super().__init__(type = "Boat")

    def move(self):
        print(f"The {self.type} is sailing ...")

class Car():

    def __init__(self, make, model, year):
        # super().__init__(type = "Car")
        self.make = make
        self.model = model
        self.year = year
        self.odometer = 0
    
    def increment_odometer(self, n=1):
        self.odometer += n

class Battery:

    def __init__(self, capacity = 10000):
        self.capacity = capacity

    def charge_battery(self, n = 10):
        self.capacity += n

class ElectricCar(Car):

    def __init__(self, make, model, year):
        super().__init__(make, model, year)
        self.battery = Battery(capacity = 10_000_000_000)

hondaEv = ElectricCar("Honda", "x", 2023)
hondaEv.increment_odometer(10)
print(hondaEv.odometer)
hondaEv.battery.charge_battery(1000)
print(hondaEv.battery.capacity)

hondaEv.move()